clang/test/Driver/sycl-offload.c

```diff
@@ -800,5 +848,79 @@
+// RUN:   | FileCheck -check-prefixes=CHK-PHASES-NO-SPLIT,CHK-PHASES-DEFAULT-MODE-NO-SPLIT %s
+// RUN:   %clang_cl -ccc-print-phases -fsycl -fsycl-targets=spir64-unknown-unknown-sycldevice-coff %s -fno-sycl-device-code-split 2>&1 \
+// RUN:   | FileCheck -check-prefixes=CHK-PHASES-NO-SPLIT,CHK-PHASES-CL-MODE-NO-SPLIT %s
+// RUN:   %clang -ccc-print-phases -target x86_64-unknown-linux-gnu -fsycl -fno-sycl-use-bitcode %s -fno-sycl-device-code-split 2>&1 \
+// RUN:   | FileCheck -check-prefixes=CHK-PHASES-NO-SPLIT,CHK-PHASES-DEFAULT-MODE-NO-SPLIT %s
+// RUN:   %clang_cl -ccc-print-phases -fsycl -fno-sycl-use-bitcode -fno-sycl-device-code-split %s 2>&1 \
+// RUN:   | FileCheck -check-prefixes=CHK-PHASES-NO-SPLIT,CHK-PHASES-CL-MODE-NO-SPLIT %s
+// RUN:   %clang -ccc-print-phases -target x86_64-unknown-linux-gnu -fsycl -fsycl-use-bitcode -fno-sycl-device-code-split %s 2>&1 \
+// RUN:   | FileCheck -check-prefixes=CHK-PHASES-NO-SPLIT,CHK-PHASES-DEFAULT-MODE-NO-SPLIT %s
+// RUN:   %clang_cl -ccc-print-phases -fsycl -fsycl-use-bitcode -fno-sycl-device-code-split %s 2>&1 \
+// RUN:   | FileCheck -check-prefixes=CHK-PHASES-NO-SPLIT,CHK-PHASES-CL-MODE-NO-SPLIT %s
+// CHK-PHASES-NO-SPLIT: 0: input, "[[INPUT:.+\.c]]", c, (host-sycl)
+// CHK-PHASES-NO-SPLIT: 1: preprocessor, {0}, cpp-output, (host-sycl)
+// CHK-PHASES-NO-SPLIT: 2: input, "[[INPUT]]", c, (device-sycl)
+// CHK-PHASES-NO-SPLIT: 3: preprocessor, {2}, cpp-output, (device-sycl)
+// CHK-PHASES-NO-SPLIT: 4: compiler, {3}, sycl-header, (device-sycl)
+// CHK-PHASES-DEFAULT-MODE-NO-SPLIT: 5: offload, "host-sycl (x86_64-unknown-linux-gnu)" {1}, "device-sycl (spir64-unknown-unknown-sycldevice)" {4}, cpp-output
+// CHK-PHASES-CL-MODE-NO-SPLIT: 5: offload, "host-sycl (x86_64-pc-windows-msvc)" {1}, "device-sycl (spir64-unknown-unknown-sycldevice-coff)" {4}, cpp-output
+// CHK-PHASES-NO-SPLIT: 6: compiler, {5}, ir, (host-sycl)
+// CHK-PHASES-NO-SPLIT: 7: backend, {6}, assembler, (host-sycl)
+// CHK-PHASES-NO-SPLIT: 8: assembler, {7}, object, (host-sycl)
+// CHK-PHASES-NO-SPLIT: 9: linker, {8}, image, (host-sycl)
+// CHK-PHASES-NO-SPLIT: 10: compiler, {3}, ir, (device-sycl)
+// CHK-PHASES-NO-SPLIT: 11: backend, {10}, assembler, (device-sycl)
+// CHK-PHASES-NO-SPLIT: 12: assembler, {11}, object, (device-sycl)
+// CHK-PHASES-NO-SPLIT: 13: linker, {12}, ir, (device-sycl)
+// CHK-PHASES-NO-SPLIT: 14: llvm-spirv, {13}, spirv, (device-sycl)
+// CHK-PHASES-NO-SPLIT: 15: clang-offload-wrapper, {14}, object, (device-sycl)
+// CHK-PHASES-DEFAULT-MODE-NO-SPLIT: 16: offload, "host-sycl (x86_64-unknown-linux-gnu)" {9}, "device-sycl (spir64-unknown-unknown-sycldevice)" {15}, image
+// CHK-PHASES-CL-MODE-NO-SPLIT: 16: offload, "host-sycl (x86_64-pc-windows-msvc)" {9}, "device-sycl (spir64-unknown-unknown-sycldevice-coff)" {15}, image
+
+// Check disabling of device code split for AOT compilation
```

Comment:
I feel like these tests can be unified with the correspondent "split" ones, since some of the phases are similar

Reply:
Then there will be **12** RUN lines for one test, are you sure?

Reply:
Valid point. Having so many checks in one `sycl-offload.c` is more of a problem - splitting the tests in a logical way would be nice someday, but it's not something this patch should be concerned with.